cob_twist_controller/src/interface_types/interface_type.cpp

```diff
@@ -0,0 +1,138 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_twist_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: June, 2015
+ *
+ * \brief
+ *   This module contains the implementation of all interface types.
+ *
+ ****************************************************************/
+
+#include "cob_twist_controller/interface_types/interface_type.h"
+
+
+InterfaceBase* InterfaceBuilder::create_interface(ros::NodeHandle& nh,
+                                                  const TwistControllerParams &params)
+{
+    InterfaceBase *ib = NULL;
+    switch(params.interface_type)
+    {
+        case VELOCITY:
+            ib = new InterfaceVelocity(nh, params);
+            break;
+        case POSITION:
+            ib = new InterfacePosition(nh, params);
+            break;
+        default:
+            ib = new InterfaceVelocity(nh, params);
+            break;
+    }
+
+    return ib;
+}
+
+/* BEGIN InterfaceVelocity ********************************************************************************************/
+inline void InterfaceVelocity::process_result(const KDL::JntArray &q_dot_ik,
```

Comment:
Method names should be camelCased (http://wiki.ros.org/CppStyleGuide#Function_.2BAC8_Methods).